connection_util.go

```diff
@@ -288,20 +288,49 @@ func populateChunkDownloader(
 	}
 }
 
-func (sc *snowflakeConn) setupOCSPPrivatelink(app string, host string) error {
-	ocspCacheServer := fmt.Sprintf("http://ocsp.%v/ocsp_response_cache.json", host)
-	logger.WithContext(sc.ctx).Debugf("OCSP Cache Server for Privatelink: %v\n", ocspCacheServer)
+func setupOCSPEnvVars(ctx context.Context, host string) error {
+	host = strings.ToLower(host)
+	if isPrivateLink(host) {
+		if err := setupOCSPPrivatelink(ctx, host); err != nil {
+			return err
+		}
+	} else if !strings.HasSuffix(host, defaultDomain) {
+		ocspCacheServer := fmt.Sprintf("http://ocsp.%v/%v", host, cacheFileBaseName)
+		logger.WithContext(ctx).Debugf("OCSP Cache Server for %v: %v\n", host, ocspCacheServer)
+		if err := os.Setenv(cacheServerURLEnv, ocspCacheServer); err != nil {
+			return err
+		}
+	} else {
+		if _, set := os.LookupEnv(cacheServerURLEnv); set {
```

Comment:
Why do we need this? Maybe some comment here?

Reply:
This is something that was introduced not by me. I think it's a measure to ensure that there will be no env var set, if there is no need for it (otherwise it could lead to driver hitting some unrelated OCSP cache servers).